src/thermo/Elements.cpp

```diff
@@ -17,135 +17,241 @@ namespace Cantera
+    {"Ru", "ruthenium",    101.07   },
+    {"Rh", "rhodium",      102.9055 },
+    {"Pd", "palladium",    106.42   },
+    {"Ag", "silver",       107.8682 },
+    {"Cd", "cadmium",      112.411  },
+    {"In", "indium",       114.82   },
+    {"Sn", "tin",          118.710  },
+    {"Sb", "antimony",     121.75   },
+    {"Te", "tellurium",    127.6    },
+    {"I",  "iodine",       126.90447},
+    {"Xe", "xenon",        131.29   },
+    {"Cs", "cesium",       132.90543},
+    {"Ba", "barium",       137.327  },
+    {"La", "lanthanum",    138.9055 },
+    {"Ce", "cerium",       140.115  },
+    {"Pr", "praseodymium", 140.90765},
+    {"Nd", "neodymium",    144.24   },
+    {"Pm", "promethium",   144.9127 },
+    {"Sm", "samarium",     150.36   },
+    {"Eu", "europium",     151.965  },
+    {"Gd", "gadolinium",   157.25   },
+    {"Tb", "terbium",      158.92534},
+    {"Dy", "dysprosium",   162.50   },
+    {"Ho", "holmium",      164.93032},
+    {"Er", "erbium",       167.26   },
+    {"Tm", "thulium",      168.93421},
+    {"Yb", "ytterbium",    173.04   },
+    {"Lu", "lutetium",     174.967  },
+    {"Hf", "hafnium",      178.49   },
+    {"Ta", "tantalum",     180.9479 },
+    {"W",  "tungsten",     183.85   },
+    {"Re", "rhenium",      186.207  },
+    {"Os", "osmium",       190.2    },
+    {"Ir", "iridium",      192.22   },
+    {"Pt", "platinum",     195.08   },
+    {"Au", "gold",         196.96654},
+    {"Hg", "mercury",      200.59   },
+    {"Tl", "thallium",     204.3833 },
+    {"Pb", "lead",         207.2    },
+    {"Bi", "bismuth",      208.98037},
+    {"Po", "polonium",     208.9824 },
+    {"At", "astatine",     209.9871 },
+    {"Rn", "radon",        222.0176 },
+    {"Fr", "francium",     223.0197 },
+    {"Ra", "radium",       226.0254 },
+    {"Ac", "actinium",     227.0279 },
+    {"Th", "thorium",      232.0381 },
+    {"Pa", "protactinium", 231.03588},
+    {"U",  "uranium",      238.0508 },
+    {"Np", "neptunium",    237.0482 },
+    {"Pu", "plutonium",    244.0482 },
 };
 
+static struct atomicWeightData isotopeWeightTable[] = {
+    {"D",  "deuterium",      2.0    },
+    {"Tr", "tritium",        3.0    },
+};
 
 doublereal LookupWtElements(const std::string& ename)
 {
-    int num = sizeof(aWTable) / sizeof(struct awData);
-    string s3 = ename.substr(0,3);
+    warn_deprecated("LookupWtElements",
+    "Use getElementWeight instead. To be removed after Cantera 2.3");
+    return getElementWeight(ename);
+}
+
+doublereal LookupWtElements(int atomicNumber)
+{
+    warn_deprecated("LookupWtElements",
+    "Use getElementWeight instead. To be removed after Cantera 2.3");
+    return getElementWeight(atomicNumber);
+}
+
+doublereal getElementWeight(const std::string& ename)
+{
+    int numElements = numElementsDefined();
+    int numIsotopes = numIsotopesDefined();
+    for (int i = 0; i < numElements; i++) {
+        if (atomicWeightTable[i].symbol.compare(0,2,ename,0,2) == 0) {
```

Comment:
In the cases of comparing to the symbol, this saves creating a variable for the substring. For the comparison with the full `ename`, using `==` would be equivalent. I guess the `compare` is more confusing, since the integer it returns for a match is 0. I'll change it back.